terraform/modules/rds/main.tf

```
@@ -0,0 +1,56 @@
+# Non-compliant RDS configuration (missing encryption, backups, and monitoring)
+resource "aws_db_instance" "main" {
+  identifier        = "${var.environment}-db"
+  engine           = "postgres"
+  engine_version   = "13.7"
+  instance_class   = var.instance_class
+  allocated_storage = 20
+
+  # Non-compliant: Credentials in plain text
+  name     = var.db_name
+  username = "admin"  # Hardcoded credentials
+  password = "insecure123!"  # Hardcoded password
+
+  # Non-compliant: Public access enabled
+  publicly_accessible = true
+
+  # Non-compliant: No encryption
+  storage_encrypted = false
+
+  # Non-compliant: Minimal backup settings
+  backup_retention_period = 0
+  skip_final_snapshot    = true
+
+  # Non-compliant: No monitoring
+  monitoring_interval = 0
+  
+  # Non-compliant: Weak security group
+  vpc_security_group_ids = [aws_security_group.db.id]
+
+  tags = {
+    Name        = "${var.environment}-db"
+    Environment = var.environment
+  }
+}
+
+# Non-compliant: Overly permissive security group
```

The RDS instance is not compliant with SOC2 requirements as it lacks encryption, sufficient backup retention, and monitoring. Additionally, it uses hardcoded credentials and allows public access, which are security risks.

```suggestion
resource "aws_db_instance" "main" {
  identifier        = "${var.environment}-db"
  engine            = "postgres"
  engine_version    = "13.7"
  instance_class    = var.instance_class
  allocated_storage = 20

  name              = var.db_name
  username          = var.db_username
  password          = var.db_password

  publicly_accessible = false

  storage_encrypted = true

  backup_retention_period = 30
  skip_final_snapshot     = false

  monitoring_interval = 60

  vpc_security_group_ids = [aws_security_group.db.id]

  tags = {
    Name        = "${var.environment}-db"
    Environment = var.environment
  }
}
```